### PR TITLE
Various nits on #129, #131

### PR DESCRIPTION
**Pull request by cabo**
Lots of niggly details re blank space.
Style consistency, some missing places (assuming we stick with the liberal approach).
Allowing "in" before or after a newline or tab.
Adding some legend to the only remaining table with XPath content in the main document.

### CODE REVIEW

File: draft-ietf-jsonpath-base.md [hunk 1]
```diff
@@ -1038,13 +1045,13 @@ regex-expr   = (path / string-literal) S regex-op S regex
...
 regex        = <TO BE DEFINED>
 
-contain-expr = containable S in-op S container
+contain-expr = containable in-op container
 containable  = path /                                 ; path to primitive value
-               number / 
+               number /
                string-literal
-in-op        = " in "                                 ; in operator
+in-op        = RS "in" RS                             ; in operator
```
Comment (goessner) [inline draft-ietf-jsonpath-base.md:1052]: yes, I was stumbeling over that also ...

File: draft-ietf-jsonpath-base.md [hunk 2]
```diff
@@ -482,14 +484,14 @@ stands for a nodelist that contains the root node of the argument,
...
-                            descendant-selector /
-                            filter-selector)
+json-path = root-selector *(S (dot-selector        /
+                               dot-wild-selector   /
+                               index-selector      /
+                               index-wild-selector /
+                               union-selector      /
+                               slice-selector      /
+                               descendant-selector /
+                               filter-selector))
```
Comment (goessner) [inline draft-ietf-jsonpath-base.md:494]: Do we really want to allow jsonpathes  scattered across several lines

```
$   .   a
[ 'b' ] 
[ 123 ] 

Has there been users demand for this ?

File: draft-ietf-jsonpath-base.md [hunk 3]
```diff
@@ -779,16 +783,19 @@ It selects elements starting at index `<start>`, ending at — but
 not including — `<end>`, while incrementing by `step`.
 
 ~~~~ abnf
-slice-selector = "[" S [start S] ":" [ S end] [S ":" S [step]] S "]"
+slice-selector = "[" S [start S] ":" S [end S] [":" S [step S]] "]"
```
Comment (goessner) [inline draft-ietf-jsonpath-base.md:786]: definitely better now ...

File: draft-ietf-jsonpath-base.md [hunk 4]
```diff
@@ -779,16 +783,19 @@ It selects elements starting at index `<start>`, ending at — but
...
 
-S              = *( %x20 / ; Space
+B              =    %x20 / ; Space
                     %x09 / ; Horizontal tab
                     %x0A / ; Line feed or New line
-                    %x0D ) ; Carriage return
+                    %x0D   ; Carriage return
+S              = *B        ; optional blank space
+RS             = 1*B       ; required blank space
+
```
Comment (goessner) [inline draft-ietf-jsonpath-base.md:798]: yes ... needied once for now ...

File: draft-ietf-jsonpath-base.md (hunk 2 above)
Comment (cabo) [reply draft-ietf-jsonpath-base.md:494]: You mean, allowing newlines in JSONPath?  Depends a bit on how long they are.  I seem to remember we wanted to be lenient here...

File: draft-ietf-jsonpath-base.md [hunk 5]
```diff
@@ -997,7 +1004,7 @@ as many times in the node list.
 The filter selector has the form `[?<expr>]`. It works via iterating over structured values, i.e. arrays and objects.
 
 ~~~~ abnf
-filter-selector    = "[?" S boolean-expr S "]"
+filter-selector    = "[" S "?" S boolean-expr S "]"
```
Comment (goessner) [inline draft-ietf-jsonpath-base.md:1007]: hmm ... consistent with blank philosophy above though

File: draft-ietf-jsonpath-base.md (hunk 2 above)
Comment (goessner) [reply draft-ietf-jsonpath-base.md:494]: yes i can remember darkly too ... so your edits all in all make sense.

**Review by goessner — APPROVED** [review]

**Pull request merged**